client/src/components/ShoeShowPage.js

```diff
@@ -60,33 +83,74 @@ const ShoeShowPage = ({ user, match }) => {
         getShoe()
     }, [])
 
-    let reviewFormComponent = ""
-    if (user){
-      reviewFormComponent = <ReviewForm postReview={postReview} shoe={shoe} errors={errors} setErrors={setErrors}/>
+  const handleVote = async (value, reviewId) => {
+    try {
+      const response = await fetch(`/api/v1/shoes/${id}/reviews/vote`, {
+        method: "POST",
+        body: JSON.stringify({ value, reviewId }),
+        headers: new Headers({
+          "Content-Type": "application/json"
+        })
+      })
+      if (!response.ok) {
+        throw new Error(
+          `${response.status} (${response.statusText})`)
+      }
+      const body = await response.json()
+      getShoe()
```

Comment:
on the initial render of the page we already fetched the data for the shoe - we should not refetch data after every time something changes, that is what state is for. After a new vote has been made, the router should send back that Vote, which we can use to update state for this Shoe show page. We want to avoid unnecessary fetch requests, especially when we already have the data here in state!